contracts/Whitelist.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.10;
+
+import "./packages/oz/Ownable.sol";
+
+/**
+ * @author Opyn Team
+ * @title Whitelist Module
+ * @notice The whitelist module keeps track of all valid Otoken contracts.
+ */
+contract Whitelist is Ownable {
+    mapping(bytes32 => bool) internal whitelistedProduct;
+
+    event ProductWhitelisted(
+        bytes32 productHash,
+        address indexed underlying,
+        address indexed strike,
+        address indexed collateral
+    );
+
+    /**
+     * @notice check if a product is supported
+     * @dev product = the hash of underlying, collateral and strike asset
+     * @param _underlying option underlying asset address
+     * @param _strike option strike asset address
+     * @param _collateral option collateral asset address
+     * @return boolean, true if product is supported
+     */
+    function isWhitelistedProduct(
+        address _underlying,
+        address _strike,
+        address _collateral
+    ) external view returns (bool) {
+        bytes32 productHash = keccak256(abi.encode(_underlying, _strike, _collateral));
+
+        return whitelistedProduct[productHash];
+    }
+
+    /**
+     * @notice whitelist a product
+     * @dev a product is the hash of the underlying, collateral and strike assets
+     * can only be called from owner address
+     * @param _underlying option underlying asset address
+     * @param _strike option strike asset address
+     * @param _collateral option collateral asset address
```

Comment:
We might need a better description for these. @zkoticha @aleone will probs have thoughts!